File: app/src/main/java/es/etologic/mahjongscoring2/app/main/combinations/CombinationsActivityModule.java
package es.etologic.mahjongscoring2.app.main.combinations;

import dagger.Module;
import dagger.Provides;
import es.etologic.mahjongscoring2.domain.use_cases.GetCombinationsUseCase;

@Module
public class CombinationsActivityModule {

    @Provides
    CombinationsViewModelFactory provideCombinationsViewModelFactory(GetCombinationsUseCase getCombinationsUseCase) {
        return new CombinationsViewModelFactory(getCombinationsUseCase);
    }
}

File: app/src/main/java/es/etologic/mahjongscoring2/app/main/combinations/CombinationsActivityModule.kt
package es.etologic.mahjongscoring2.app.main.combinations

import dagger.Module
import dagger.Provides
import es.etologic.mahjongscoring2.domain.use_cases.GetCombinationsUseCase

@Module
class CombinationsActivityModule {
    
    @Provides
    internal fun provideCombinationsViewModelFactory(getCombinationsUseCase: GetCombinationsUseCase): CombinationsViewModelFactory {
        return CombinationsViewModelFactory(getCombinationsUseCase)
    }
}
